cmake/configure/configure_1_cuda.cmake

```diff
@@ -25,29 +25,60 @@ SET(FEATURE_CUDA_DEPENDS CXX11)
 
 MACRO(FEATURE_CUDA_FIND_EXTERNAL var)
 
-  #
-  # FIXME Restructure this call into a ./modules/FindCUDA.cmake file and
-  # use DEAL_II_PACKAGE_HANDLE
-  #
-  FIND_PACKAGE(CUDA)
-  MARK_AS_ADVANCED(
-    CUDA_HOST_COMPILER
-    CUDA_SDK_ROOT_DIR
-    CUDA_TOOLKIT_ROOT_DIR
-    CUDA_USE_STATIC_CUDA_RUNTIME
-    )
+  IF(${DEAL_II_WITH_CUDA})
+    #
+    # FIXME Restructure this call into a ./modules/FindCUDA.cmake file and
+    # use DEAL_II_PACKAGE_HANDLE
+    #
+    FIND_PACKAGE(CUDA)
+    MARK_AS_ADVANCED(
+      CUDA_HOST_COMPILER
+      CUDA_SDK_ROOT_DIR
+      CUDA_TOOLKIT_ROOT_DIR
+      CUDA_USE_STATIC_CUDA_RUNTIME
+      )
 
-  IF(CUDA_FOUND)
+    IF(CUDA_FOUND)
 
-    SET(CUDA_ATTACH_VS_BUILD_RULE_TO_CUDA_FILE FALSE)
+      IF("${CUDA_NVCC_FLAGS}" MATCHES "-arch")
 
-    SET(CUDA_NVCC_FLAGS ${CUDA_NVCC_FLAGS} -std=c++11)
+        # Compute Capability specified explicitly.
+        # Now parse:
 
-    # FIXME: CUDA compiler NVCC doesn't support C++14.
+        IF("${CUDA_NVCC_FLAGS}" MATCHES "-arch[ ]*sm_([0-9]*)")
+          SET(CUDA_COMPUTE_CAPABILITY "${CMAKE_MATCH_1}")
+        ELSEIF("${CUDA_NVCC_FLAGS}" MATCHES "-arch=sm_([0-9]*)")
+          SET(CUDA_COMPUTE_CAPABILITY "${CMAKE_MATCH_1}")
+        ELSE()
+          MESSAGE( FATAL_ERROR "Missformed Compute Capability specified")
+        ENDIF()
+
+        IF("${CUDA_COMPUTE_CAPABILITY}" LESS "35")
+          MESSAGE( FATAL_ERROR "Too low CUDA Compute Capability specified -- deal.II requires at least 3.5 ")
```

Comment:
This means that the user explicitly said that we should compile using something that we don't support, it is similar to specifying a C++ standard that is too old. What is the sensible fallback? I think the configuration should not carry on using some different albeit correct configuration, since that would create unnecessary confusion as to what configuration was actually use.